src/IdentityServer/Configuration/DependencyInjection/Options/DynamicProviderOptions.cs

```diff
@@ -68,14 +69,14 @@ public class DynamicProviderType
         /// <summary>
         /// The type of the handler.
         /// </summary>
-        public Type HandlerType { get; set; }
+        public Type HandlerType { get; set; } = default!;
```

Comment:
For most of those, I looked in our code base how they were being used and wherever there was no check for null, I marked the properties as such.

Reply:
That's fair - I don't think we want to change our code internally to add a bunch of redundant null checks. But I'd still like to make the tooling do as much for the user as it can. Maybe what we should do is keep these `=default!`s and then when we upgrade replace those with required properties.

Reply:
Sure, we can look at those in vNext